Dockerfile

```diff
@@ -58,14 +50,13 @@ RUN cd /tmp \
     && rm -rf /tmp/bats-core
 
 # install criu
-ARG CRIU_VERSION
-RUN mkdir -p /usr/src/criu \
-    && curl -fsSL https://github.com/checkpoint-restore/criu/archive/${CRIU_VERSION}.tar.gz | tar -C /usr/src/criu/ -xz --strip-components=1 \
-    && cd /usr/src/criu \
-    && echo 1 > .gitid \
-    && make -j $(nproc) install-criu \
-    && cd - \
-    && rm -rf /usr/src/criu
+RUN echo 'deb https://download.opensuse.org/repositories/devel:/tools:/criu/Debian_10/ /' > /etc/apt/sources.list.d/criu.list \
+    && wget -nv https://download.opensuse.org/repositories/devel:/tools:/criu/Debian_10/Release.key -O- | sudo apt-key add - \
+    && apt-get -q update \
+    && apt-get -q install -y --no-install-recommends criu \
```

Comment:
can we apt get `crui` and `skopeo` at once?

Reply:
Yeah, I was thinking about that. It will be faster to merge all the apt-related steps into one, iow add all repos and run `apt update && apt install`.

The biggest issue with the current Dockerfile is is downloads apt metadata 3 times and then removes it.

I took a look at moby/moby and in there they added a cache so the stages are still separate but the metadata is not downloaded multiple times. This needs a recent docker though, and afaik do not work with podman (it does not error out, but no caching is done either).

Given the fact that this Dockerfile is pretty simple, I think combining all the steps is the best solution.

Reply:
Updated.